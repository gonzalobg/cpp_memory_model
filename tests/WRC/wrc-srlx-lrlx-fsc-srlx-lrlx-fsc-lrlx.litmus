C wrc-srlx-lrxl-fsc-srlx-lrlx-fsc-lrlx.litmus
{ [x] = 0; [y] = 0;  }

P0 (int* x) {
  atomic_store_explicit(x, 1, memory_order_relaxed);
}

P1 (int* x, int* y) {
  int a = atomic_load_explicit(x, memory_order_relaxed);
  if (a == 1) {
    atomic_thread_fence(memory_order_seq_cst);  
    atomic_store_explicit(y, 1, memory_order_relaxed);
  }
}

P2 (int* x, int* y) {
  int b = atomic_load_explicit(y, memory_order_relaxed);
  if (b == 1) {
     atomic_thread_fence(memory_order_seq_cst);  
     int c = atomic_load_explicit(x, memory_order_relaxed);
  }
}

~exists(1:a=1 /\ 2:b=1 /\ 2:c=0)