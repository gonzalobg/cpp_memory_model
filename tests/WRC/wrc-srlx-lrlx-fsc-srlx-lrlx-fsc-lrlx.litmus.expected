Test wrc-srlx-lrxl-fsc-srlx-lrlx-fsc-lrlx Forbidden
States 3
1:a=0; 2:b=0; 2:c=0;
1:a=1; 2:b=0; 2:c=0;
1:a=1; 2:b=1; 2:c=1;
Ok
Witnesses
Positive: 3 Negative: 0
Condition ~exists (1:a=1 /\ 2:b=1 /\ 2:c=0)
Observation wrc-srlx-lrxl-fsc-srlx-lrlx-fsc-lrlx Never 0 3
Hash=e63930180dc1e1c47a935bbf84278374

